pkg/vault/instances.go

```diff
@@ -245,60 +271,68 @@ func createClient(addr string,
 	config.Address = addr
 	client, err := api.NewClient(config)
 	if err != nil {
-		log.WithError(err)
-		fmt.Println(fmt.Sprintf("Failed to initialize Vault client for %s", addr))
-		fmt.Println(fmt.Sprintf("SKIPPING ALL RECONCILIATION FOR: %s\n", addr))
-		return // skip entire reconcilation for this instance
+		log.WithError(err).Errorf("[Vault Client] failed to initialize Vault client for `%s`", addr)
+		log.Warnf("SKIPPING ALL RECONCILIATION FOR: %s", addr)
+		return // Skip entire reconciliation for this instance.
 	}
 
 	// indicates kube auth should be utilized
 	if len(bundle.KubeRoleName) > 0 {
 		err := configureKubeAuthClient(client, bundle)
 		if err != nil {
-			log.WithError(err)
-			fmt.Println(fmt.Sprintf("[Vault Client] failed to login to %s with kube sa token", addr))
-			fmt.Println(fmt.Sprintf("SKIPPING ALL RECONCILIATION FOR: %s\n", addr))
-			return // skip entire reconcilation for this instance
+			log.WithError(err).Errorf("[Vault Client] failed to login to `%s` with kube sa token", addr)
+			log.Warnf("SKIPPING ALL RECONCILIATION FOR: %s", addr)
+			return // Skip entire reconciliation for this instance.
 		}
 	} else {
 		accessCreds := make(map[string]string)
 		for _, cred := range bundle.VaultSecrets {
 			// masterAddress hard-coded because all "child" vault access credentials must be pulled from master
 			processedCred, err := GetVaultSecretField(masterAddress, cred.Path, cred.Field, bundle.SecretEngine)
 			if err != nil {
-				log.WithError(err).Fatal()
+				log.WithError(err).Fatal("[Vault Client] unable to retrieve credentials from master Vault")
```

Comment:
**To the reviewer:**

The function `GetVaultSecretField()` called above is used to return more complex error messages. However, it's preferable to simply return a more plain error to be decorated later (or wrapped, if you wish). Thus, we move the more specific error message, also more helpful to the user, here since the service will exist here as it's a `Fatal()` log.